ldelf/ta_elf.c

```diff
@@ -855,6 +861,56 @@ static void populate_segments(struct ta_elf *elf)
 	}
 }
 
+static void parse_property_segment(struct ta_elf *elf)
+{
+	char *desc = NULL;
+	size_t align = elf->prop_align;
+	size_t desc_offset = 0;
+	size_t prop_offset = 0;
+	vaddr_t va = elf->load_addr + elf->prop_start;
+	Elf_Note *note = (void *)va;
+	char *name = (char *)(note + 1);
+
+	if (elf->prop_memsz < sizeof(*note) + sizeof(ELF_NOTE_GNU))
+		return;
+
+	if (note->n_type != NT_GNU_PROPERTY_TYPE_0 ||
+	    note->n_namesz != sizeof(ELF_NOTE_GNU) ||
+	    memcmp(name, ELF_NOTE_GNU, sizeof(ELF_NOTE_GNU)))
+		return;
+
+	desc_offset = ROUNDUP(sizeof(*note) + sizeof(ELF_NOTE_GNU), align);
```

Comment:
`ROUNDUP()` produces an unexpected result of 2nd argument is not a power of 2. Should test `IS_POWER_OF_TWO(align)`.

Reply:
Thanks...will add